packages/lit-element/src/lib/updating-element.ts

```diff
@@ -482,66 +443,13 @@ export abstract class UpdatingElement extends HTMLElement {
       : undefined;
   }
 
-  /**
-   * Returns true if a property should request an update.
-   * Called when a property value is set and uses the `hasChanged`
-   * option for the property if present or a strict identity check.
-   * @nocollapse
-   */
-  private static _valueHasChanged(
-    value: unknown,
-    old: unknown,
-    hasChanged: HasChanged = notEqual
-  ) {
-    return hasChanged(value, old);
-  }
-
-  /**
-   * Returns the property value for the given attribute value.
-   * Called via the `attributeChangedCallback` and uses the property's
-   * `converter` or `converter.fromAttribute` property option.
-   * @nocollapse
-   */
-  private static _propertyValueFromAttribute(
-    value: string | null,
-    options: PropertyDeclaration
-  ) {
-    const type = options.type;
-    const converter = options.converter || defaultConverter;
-    const fromAttribute =
-      typeof converter === 'function' ? converter : converter.fromAttribute;
-    return fromAttribute ? fromAttribute(value, type) : value;
-  }
-
-  /**
-   * Returns the attribute value for the given property value. If this
-   * returns undefined, the property will *not* be reflected to an attribute.
-   * If this returns null, the attribute will be removed, otherwise the
-   * attribute will be set to the value.
-   * This uses the property's `reflect` and `type.toAttribute` property options.
-   * @nocollapse
-   */
-  private static _propertyValueToAttribute(
-    value: unknown,
-    options: PropertyDeclaration
-  ) {
-    if (options.reflect === undefined) {
-      return;
-    }
-    const type = options.type;
-    const converter = options.converter;
-    const toAttribute =
-      (converter && (converter as ComplexAttributeConverter).toAttribute) ||
-      defaultConverter.toAttribute;
-    return toAttribute!(value, type);
-  }
-
-  private _updateState!: UpdateState;
-  private _instanceProperties?: PropertyValues;
+  private _instanceProperties?: PropertyValues = new Map();
   // Initialize to an unresolved Promise so we can make sure the element has
   // connected before first update.
   private _updatePromise!: Promise<unknown>;
-  private _enableUpdatingResolver: (() => void) | undefined;
+
+  protected isUpdatePending = false;
+  protected hasUpdated = false;
```

Comment:
In the previous version, I believe there was a value placed on saving instance memory (i.e. state bitmap, not initializing rarely used instance fields like `_instanceProperties`).

Are the changes here turning the dial from optimizing memory to optimizing code? If so, probably worth stating in the PR goals and document that as an explicit decision?

Reply:
Yes, good point, will update PR.